js/common/context.go

```diff
@@ -48,12 +73,20 @@ func GetRuntime(ctx context.Context) *goja.Runtime {
 }
 
 // WithInitEnv attaches the given init environment to the context.
+//
+// Deprecated: Implement the modules.VU interface for sharing the init environment.
 func WithInitEnv(ctx context.Context, initEnv *InitEnvironment) context.Context {
 	return context.WithValue(ctx, ctxKeyInitEnv, initEnv)
 }
 
 // GetInitEnv retrieves the attached init environment struct from the given context.
+//
+// Deprecated: Use modules.VU for get the init environment.
 func GetInitEnv(ctx context.Context) *InitEnvironment {
+	logger := logger(ctx)
```

Comment:
should we check and skip in case of `logger == nil`? It would be a bug but maybe avoid us to hotfix it